scummvm/scummvm
# SCUMM: Make Dig crypt subtitle workaround an optional waitForActor()

## Description

PR #3014 and commit 3b05a52588fc92885d7cae0c96be7eb516d27e5c added a workaround for an original subtitle issue in The Dig, which also happens with the original interpreters. See [Trac#4410](https://bugs.scummvm.org/ticket/4410).

It was decided to implement this workaround in `actor.cpp`, so that Low wouldn't wait for the end of his sentence before moving to the crypt.

The problem with this is that Low's walking animation becomes very strange:

https://user-images.githubusercontent.com/9024526/194424291-c054a7e2-65ad-42a3-b9f1-4846e9d3c70c.mov

(note that he also loses his lipsyncing).

So this PR just changes this for a `waitForActor()` emulation, as we already do for most of these cases.

Having him wait finishing his sentence before he moves to the crypt may feel a bit odd too, but Low already does that in some other parts of the game, and he just reacts at a very slow pace in general:

https://user-images.githubusercontent.com/9024526/194424755-4afbdfa2-a1f2-40ae-b3ea-1e2a870804ad.mov

Moving this to `o6_talkActor()` also means that it's next to the other `waitForActor()` workarounds, where we try to only target the impacted lines when we can.

Here's the part of the `roomobj-58-402.dmp` script triggering this line, for reference:

```
[0018] (43) var253 = 45
[001E] (65) stopObjectCodeA()
[001F] (5D) if (bitvar94 && (getState(388) == 2)) {
[002E] (5D)   if (var269 == 3) {
[0038] (0C)     dup[1] = localvar0
[003C] (5D)     if (dup[1] == 0) {
[0043] (5D)       if (bitvar78) {
[004A] (B3)         stopSentence()
[004B] (5D)         if (!bitvar97) {
[0052] (BA)           talkActor("/TOMB.022/Now that I know what I'm looking for, maybe I can get this slab out of the way.",3)
[00B0] (**)         }
[00B0] (5E)         startScript(0,2005,[])
[00BA] (7C)         stopScript(0)
[00BE] (**)       }
[00BE] (73)       /* jump c2; */
[00C1] (**)     }
[00C1] (**)   }
[00C2] (**) }
...
```

This PR also makes this workaround use `_enableEnhancements` while there.

## Test

1. Download the savegame at the top of [Trac#4410](https://bugs.scummvm.org/ticket/4410)
2. Load it
3. Click on the statue so that it vanishes
4. Exit the tomb
5. Enter inside the tomb again
6. Click on the crypt in the middle of the room

Whether you have speech/subtitles, the "Now that I know what I'm looking for…" line should be said/printed. Other interactions with this object in this room appear to be fine, including at earlier stages when the puzzles in this room haven't been solved yet.

Tested with the DOS/English and DOS/French versions.
